src/boost_histogram/_internal/hist.py

```diff
@@ -195,9 +195,14 @@ def fill(self, *args, **kwargs):
             threaded filling.  Using 0 will automatically pick the number of
             available threads (usually two per core).
         """
-
+        # should not allow float weight for int storage
+        if "weight" in kwargs:
+                for w in kwargs["weight"]:
+                    if not isinstance(self._storage._PyType, type(w)):
```

Comment:
This is an incorrect check; weight could be a single value or an array of values, this is only checking for one int rather than an array (I think).

Reply:
Interesting, it does work on arrays of ints? Something like this might fix the other PR.

Reply:
You might misunderstand. As you can see, `for w in kwargs["weight"]:` will iterate every value in weight list. And you can see the demo above in my comments. Every possible broken case is considered.

Reply:
Ah, okay. I see the loop. Does this work when this is not an array? And this will be horribly slow, since it's a Python loop.

Reply:
You get the point! Only `weight=[1]` is supported now. `weight=1` is not supported. But it would not be a hard job.

Yep, I have seen the importance to deep into the underlying C++ parts.